spec/app/domain/authentication/authn-gcp/extract_resource_restrictions_spec.rb

```diff
@@ -0,0 +1,440 @@
+  let(:mocked_role_class_return_annotations_list_with_duplication) {double("RoleClass")}
+
+  before(:each) do
+    allow(mocked_resource_class_return_nil).to receive(:[])
+                                                 .with(any_args)
+                                                 .and_return(nil)
+
+
+    allow(mocked_resource_class_return_annotations_list).to receive(:[])
+                                                              .with(any_args)
+                                                              .and_return(mocked_role_class_return_annotations_list)
+    allow(mocked_role_class_return_annotations_list).to receive(:annotations)
+                                                          .and_return(annotations_list)
+
+    allow(mocked_resource_class_return_annotations_nil).to receive(:[])
+                                                             .with(any_args)
+                                                             .and_return(mocked_role_class_return_annotations_nil)
+    allow(mocked_role_class_return_annotations_nil).to receive(:annotations)
+                                                         .and_return(nil)
+
+    allow(mocked_resource_class_return_annotations_empty_list).to receive(:[])
+                                                                    .with(any_args)
+                                                                    .and_return(mocked_role_class_return_annotations_empty_list)
+    allow(mocked_role_class_return_annotations_empty_list).to receive(:annotations)
+                                                                .and_return([])
+
+    allow(mocked_resource_class_return_annotations_list_with_duplications).to receive(:[])
+                                                                                .with(any_args)
+                                                                                .and_return(mocked_role_class_return_annotations_list_with_duplication)
+    allow(mocked_role_class_return_annotations_list_with_duplication).to receive(:annotations)
+                                                                           .and_return(annotations_list_with_duplications)
+
+
+  end
+
+  #  ____  _   _  ____    ____  ____  ___  ____  ___
+  # (_  _)( )_( )( ___)  (_  _)( ___)/ __)(_  _)/ __)
+  #   )(   ) _ (  )__)     )(   )__) \__ \  )(  \__ \
+  #  (__) (_) (_)(____)   (__) (____)(___/ (__) (___/
+
+  context "An invalid input parameters to extract resource restrictions" do
+    context "when account not exists" do
+      subject do
+        Authentication::AuthnGCP::ExtractResourceRestrictions.new(
+          resource_class: mocked_resource_class_return_nil
+        ).call(
+          account:           invalid_account,
+          username:          valid_host,
+          extraction_prefix: valid_prefix
+        )
+      end
+
+      it "raises an error" do
+        expect {subject}.to raise_error(Errors::Authentication::Security::RoleNotFound)
```

Comment:
this is not a healthy behaviour. if the account does not exist i would expect an `AccountNotDefined` error

Reply:
good point i will add an account validation to the logic